src/components/Timer/Timer.tsx

```diff
@@ -0,0 +1,53 @@
+import ReactDOM from "react-dom";
+import {useEffect, useState} from "react";
+import Parse from "parse";
+import classNames from "classnames";
+import store, {useAppSelector} from "store";
+import {ActionFactory} from "store/action";
+import {ReactComponent as CloseIcon} from "assets/icon-close.svg";
+import "./Timer.scss";
+
+type TimerProps = {
+  endTime: Date;
+};
+
+export const Timer = (props: TimerProps) => {
+  const calculateTime = () => {
+    const difference = +props.endTime - +new Date();
+    return {
+      h: Math.max(Math.floor((difference / 1000 / 60 / 60) % 24), 0),
+      m: Math.max(Math.floor((difference / 1000 / 60) % 60), 0),
+      s: Math.max(Math.floor((difference / 1000) % 60), 0),
+    };
+  };
+
+  const isModerator = useAppSelector((state) => state.users.admins.some((user) => user.id === Parse.User.current()?.id));
+  const [timeLeft, setTimeLeft] = useState<{h: number; m: number; s: number}>(calculateTime());
+
+  useEffect(() => {
+    const timerUpdateTimeout = setTimeout(() => {
+      setTimeLeft(calculateTime());
+    }, 250);
+    return () => clearTimeout(timerUpdateTimeout);
+  });
+
+  return ReactDOM.createPortal(
+    <aside
+      className={classNames(
+        "timer",
+        {"timer--expired": timeLeft.m === 0 && timeLeft.s === 0},
+        {"timer--top": document.getElementById("menu-bars")?.classList.contains("menu-bars--bottom")}
```

Comment:
This could cause problems, because the timer and the bar overlap (only for a short period of time) if we update the bottom bar first.